### DISCUSSION TITLE
Welcome to Cyfrin Updraft Discussions!

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [Cyfrin/Updraft#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by fuzzylogic102** [comment]
Hello everyone, I've been taking this amazing course for some time now. So far so good, I currently hit a problem with a line of code in the Interactions.s.sol file `fundMeInstance.fund{value: SEND_VALUE}();`, this line of code in the fundFundMe function reverts with an OutOfFunds error. Although I double-checked that the SEND_VALUE (0.01 ether), is being declared correctly and is more than the MIN requirement for the fund function. The SEND_VALUES are also the same in Intreractions.s.sol and IntreractionsTest.t.sol. I've been trying to debug this problem forever but just couldn't pinpoint the problem. `

This is the function in Interactions.s.sol:

```
function fundFundMe(address mostRecentlyDeployed) public {
        console.log("Funding address:", mostRecentlyDeployed); // Log the address being funded
        console.log("Creating Instance");
        FundMe fundMeInstance = FundMe(payable(mostRecentlyDeployed));
        uint256 balance = address(fundMeInstance).balance;
        console.log(balance);
        console.log("Funding the instance");
        console.log(SEND_VALUE);
        fundMeInstance.fund{value: SEND_VALUE}();
    }
``` 

This is the function in InteractionTest.t.sol:       

```
function testUserCanFundInteractions() public {
        FundFundMe fundFundMe = new FundFundMe();
        address fundMeAddress = address(fundMe); // Use the correct contract address

        console.log("Funding contract address:", fundMeAddress);
        console.log("Starting balance of contract:", address(fundMeAddress).balance);

        // Prank as the user before calling the fund function
        fundFundMe.fundFundMe(fundMeAddress);

        console.log("Ending balance of contract:", address(fundMeAddress).balance);

        address funder = fundMe.getFunder(0);
        assertEq(funder, USER);
    }
```


and this is the fundMe function:

```
function fund() public payable {
    console.log("IM BEING CALLED NOW");
    uint256 conversionRate = msg.value.getConversionRate(s_priceFeed);
    require(
        conversionRate >= MINIMUM_USD,
        "You need to spend more ETH!"
    );
    s_addressToAmountFunded[msg.sender] += msg.value;
    s_funders.push(msg.sender);
}
```
This is the error output with verbosity:

```
Traces:
  [1082162] InteractionTest::testUserCanFundInteractions()
    ├─ [1023965] → new FundFundMe@0x2e234DAe75C793f67A35089C9d99245E1C58470b
    │   └─ ← [Return] 5004 bytes of code
    ├─ [0] VM::prank(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← [Return] 
    ├─ [0] VM::deal(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], 10000000000000000000 [1e19])
    │   └─ ← [Return] 
    ├─ [0] console::log("Funding contract address:", FundMe: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496]) [staticcall]
    │   └─ ← [Stop] 
    ├─ [0] console::log("Starting balance of contract:", 0) [staticcall]
    │   └─ ← [Stop] 
    ├─ [10401] FundFundMe::fundFundMe(FundMe: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496])
    │   ├─ [0] console::log("Funding address:", FundMe: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496]) [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] console::log("Creating Instance") [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] console::log(0) [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] console::log("Funding the instance") [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] console::log(100000000000000000000 [1e20]) [staticcall]
    │   │   └─ ← [Stop] 
    │   ├─ [0] FundMe::fund{value: 100000000000000000000}()
    │   │   └─ ← [OutOfFunds] EvmError: OutOfFunds
    │   └─ ← [Revert] EvmError: Revert
    └─ ← [Revert] EvmError: Revert
```

Any help is appreciated, I just want to fix this and continue this amazing course.

**Reply by EngrPips** [reply]
Hello @fuzzylogic102 Please open a separate and appropriate discussion for this issue on the official Github https://github.com/Cyfrin/foundry-full-course-cu/discussions of this course.

**Comment by Legend1nz** [comment]
i have followed your step gradually but i don't know why it is showing me this error
where am i getting it wrong?
look at the picture
![image](https://github.com/user-attachments/assets/6d42d886-c78e-4d94-a2f1-04005ead8117)

**Reply by EngrPips** [reply]
Hello @Legend1nz, Please open a separate and appropriate discussion for this issue on the official Github https://github.com/Cyfrin/foundry-full-course-cu/discussions of this course.

**Comment by selkoyuncu** [comment]
Hello ,

i have a conceptual question .
From the a youtube video i watch, it has been highlighted that blockchains come concuss only on 3 things by nature, these are ;

1- digitalkey signatures ,
2- token ledgers,
3- state of the blockchain

 I am not clear about what is state of the blockchain . does it mean, after BFTs work, the snapshot of chains agreed by all validators/miners or what it is ?  Or is it sth  For example, if a smart contract has a variable "x = 10" at the beginning and transactions update it (e.g., increment by 5), the state reflects "x = 15" after processing those transactions. keeping these data between all nodes ?

PS: the video is from bankless interview with sergey. and he is mentining that rather than above items, oracles are capable to do.

**Reply by EngrPips** [reply]
What do you mean `BFTs work`?

**Comment by MrAgarwal2002** [comment]


**Reply by EngrPips** [reply]
Please open a separate discussion for this.

**Comment by DavidNg-DAO** [comment]
HI i am new here just join Cyfin Updraft course thanks

**Reply by EngrPips** [reply]
Welcome aboard, Friend. We are glad to have you.
